### ISSUE TITLE
Services with multiple ports mixing up in maesh

### ISSUE THREAD

**Issue by tornyairoland** [containous/maesh#423]
Hi!

We have a Service that has two ports: port 80 is used from inside the cluster, port 8080 is accessed externally.
Using maesh and calling the service with the `.maesh` DNS name and port 80, the request sometimes gets routed to port 8080 of the service, which causes problems.
We found that for each of the maesh-created services have both endpoints, are they supposed to be in both of them? If so, how does maesh decide which one to route to?

Service configuration:
```json
{
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
        "name": "dliver-direct-message-service",
        "namespace": "sbx"
    },
    "spec": {
        "clusterIP": "172.20.219.199",
        "ports": [
            {
                "name": "api",
                "port": 80,
                "protocol": "TCP",
                "targetPort": 80
            },
            {
                "name": "public",
                "port": 8080,
                "protocol": "TCP",
                "targetPort": 8080
            }
        ],
        "selector": {
            "app": "dliver-direct-message-service"
        },
        "type": "ClusterIP"
    },
    "status": {
        "loadBalancer": {}
    }
}
```

Maesh service configuration:
```json
{
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
        "labels": {
            "app": "maesh"
        },
        "name": "infra-dliver-direct-message-service-6d61657368-sbx",
        "namespace": "infra",
    },
    "spec": {
        "clusterIP": "172.20.243.188",
        "ports": [
            {
                "name": "api",
                "port": 80,
                "protocol": "TCP",
                "targetPort": 5000
            },
            {
                "name": "public",
                "port": 8080,
                "protocol": "TCP",
                "targetPort": 5001
            }
        ],
        "selector": {
            "component": "maesh-mesh"
        },
        "type": "ClusterIP"
    },
    "status": {
        "loadBalancer": {}
    }
}
```

Controller configuration from the controller API:
```javascript
{
  "http": {
    "routers": {
       // ...
      "dliver-dir-sbx-80-0504adbb248e7344": {
        "entryPoints": [
          "http-5000"
        ],
        "service": "dliver-dir-sbx-80-0504adbb248e7344",
        "rule": "Host(`dliver-direct-message-service.sbx.maesh`) || Host(`172.20.219.199`)"
      },
      "dliver-dir-sbx-8080-20e3ccec1dd7334b": {
        "entryPoints": [
          "http-5001"
        ],
        "service": "dliver-dir-sbx-8080-20e3ccec1dd7334b",
        "rule": "Host(`dliver-direct-message-service.sbx.maesh`) || Host(`172.20.219.199`)"
      },
      // ...
    },
    "services": {
        // ...
      "dliver-dir-sbx-80-0504adbb248e7344": {
        "loadBalancer": {
          "servers": [
            {
              "url": "http://10.152.1.53:8080"
            },
            {
              "url": "http://10.152.1.53:80"
            }
          ],
          "passHostHeader": true
        }
      },
      "dliver-dir-sbx-8080-20e3ccec1dd7334b": {
        "loadBalancer": {
          "servers": [
            {
              "url": "http://10.152.1.53:8080"
            },
            {
              "url": "http://10.152.1.53:80"
            }
          ],
          "passHostHeader": true
        }
      },
      // ...
    }
  },
  "tcp": {}
}
```
Kubernetes version: v1.14.9-eks-c0eccc
Maesh version: v1.1.0-rc1

**Comment by dtomcej** [comment]
Thanks for this report. Will try to repro, and will reply back on this ticket with my results.

**Comment by dtomcej** [comment]
Hi @tornyairoland,

Thanks for the great bug report. Was super easy to reproduce your issue from your objects and API configuration report.

I have created a PR to fix it, and will be included in the next RC release.

**Comment by traefiker** [comment]
Closed by #428.
